rancher/rancher
# Support Rancher Pay-As-You-Go (PAYG) offering

Enhance the supportconfig functionality to support Rancher PAYG offering in addition to the existing Bring-Your-Own (BYO) offering, by checking the the existence of the new CSP billing adapter rancher-csp-billing-adapter.

## Issue: None
 
## Problem
Enhance the supportconfig functionality to support Rancher PAYG offering.
 
## Solution
Supportconfit API will be supporting PAYG offering by checking for the existence of the new CSP billing adapter rancher-csp-billing-adapter. If it exist, it will have precedence over the existing Bring-Your-Own (BYO) CSP adapter.
 
## Testing
Testing is done manually by installing the new CSP billing adapter helm chart, similar to how it was done with the old CSP adapter helm chart, and observe that supportconfig API is working properly.

## Engineering Testing
### Manual Testing
1. install the new CSP billing adapter helm chart (i.e. `helm install rancher-csp-billing-adapter rancher-csp-billing-adapter --create-namespace --namespace cattle-csp-billing-adapter-system`)
2. call the `/v1/generateSUSERancherSupportConfig` API to make sure it is returning the support config tarball.

### Automated Testing
TBD

Summary:   Enhance the supportconfig functionality to support Rancher PAYG offering in addition to the existing Bring-Your-Own (BYO) offering, by checking for the existence of the new CSP billing adapter rancher-csp-billing-adapter.

## QA Testing Considerations
TBD
 
### Regressions Considerations
Make sure the changes did not break the existing BYO (CSP adapter) offering.

Existing / newly added automated tests that provide evidence there are no regressions:
* _TODO_
